cmake/modules/AddSwiftHostLibrary.cmake

```diff
@@ -56,7 +56,7 @@ function(add_swift_host_library name)
       -emit-module-interface-path;${module_interface_file}
       >)
 
-  if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  if(SWIFT_DRIVER_ENABLE_WMO)
```

Comment:
We don't want it enabled in debug, so seems like we'd still want the old check as well.

Reply:
How about sinking that into the default value check?

Reply:
Sure, works for me 👍

Reply:
Discussed offline and settled on `SWIFT_SYNTAX_ENABLE_WMO_PRE_3_26`.  It was always meant to be `SWIFT_SYNTAX_ENABLE_WMO` but my brain wasn't braining very well.